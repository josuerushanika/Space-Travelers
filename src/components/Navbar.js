import { NavLink } from 'react-router-dom';
import '../style/Navbar.css';

const Navbar = () => (
  <div>
    <ul className="navs">
      <li><NavLink className="navLink" to="/">Rockets</NavLink></li>
      <li><NavLink className="navLink" to="/Missions">Missions</NavLink></li>
      <li><NavLink className="navLink" to="/Profile">Profile</NavLink></li>
    </ul>

  </div>
);

export default Navbar;
