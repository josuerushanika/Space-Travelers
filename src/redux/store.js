import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './mission/missionSlice';
import rocketReducer from './rockets/rocketSlice';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
    missions: missionReducer,
  },
});

export default store;

