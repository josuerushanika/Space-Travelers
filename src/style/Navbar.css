.navBox {
  display: flex;
  justify-content: space-between;
  padding: 0% 2%;
}

.navleft {
  display: flex;
  width: 40%;
}

h1 {
  margin-left: 2%;
}

.navright {
  display: flex;
  align-items: center;
  width: 17.9%;
}

.navs {
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  list-style: none;
  padding: 0%;
}

.navLink {
  text-decoration: none;
  align-self: center;
  font-size: 1.2rem;
}

.line {
  color: blue;
}

.profile {
  margin-left: -8%;
}

.navLink:focus {
  color: red;
  text-decoration: underline;
}

.page-logo {
  height: 70px;
  width: auto;
}
