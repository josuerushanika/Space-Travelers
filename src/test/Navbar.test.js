import { render, screen } from '@testing-library/react';
import { MemoryRouter, BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';
import store from '../redux/store';

describe('Navbar', () => {
  it('renders Navbar', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Navbar />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Navbar component', () => {
    render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );

    const navElement = screen.getByRole('navigation');
    expect(navElement).toBeInTheDocument();

    const logoElement = screen.getByAltText('page-logo');
    expect(logoElement).toBeInTheDocument();

    const rocketsLink = screen.getByText('Rockets');
    expect(rocketsLink).toBeInTheDocument();
    expect(rocketsLink).toHaveAttribute('href', '/');

    const missionsLink = screen.getByText('Missions');
    expect(missionsLink).toBeInTheDocument();
    expect(missionsLink).toHaveAttribute('href', '/Missions');

    const profileLink = screen.getByText('Profile');
    expect(profileLink).toBeInTheDocument();
    expect(profileLink).toHaveAttribute('href', '/Profile');
  });
});
