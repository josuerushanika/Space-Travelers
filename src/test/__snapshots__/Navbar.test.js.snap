// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders Navbar 1`] = `
<div
  className="navBox"
  role="navigation"
>
  <img
    alt="page-logo"
    className="page-logo"
    src="icon.png"
  />
  <ul
    className="navs"
  >
    <li>
      <a
        aria-current="page"
        className="navLink active"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        className="navLink"
        href="/Missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <li>
      <a
        className="navLink"
        href="/Profile"
        onClick={[Function]}
      >
        Profile
      </a>
    </li>
  </ul>
</div>
`;
